src/Uno.UI/Controls/UIFontHelper.iOS.cs

```diff
@@ -64,9 +64,7 @@ private static nfloat GetScaledFontSize(nfloat size, nfloat? basePreferredSize)
 				return size;
 			}
 
-			var scaledSize = size * (basePreferredSize / DefaultUIFontPreferredBodyFontSize) ?? (float)1.0;
-			return scaledSize > FeatureConfiguration.Font.MaximumTextScaleFactor ?
-				FeatureConfiguration.Font.MaximumTextScaleFactor : scaledSize;
+			return size * Math.Min((basePreferredSize / DefaultUIFontPreferredBodyFontSize),FeatureConfiguration.Font.MaximumTextScaleFactor) ?? (float)1.0;
```

Comment:
Let's use this, to avoid nullability conflicts, and for readability:

```suggestion
            var originalScale = (basePreferredSize / DefaultUIFontPreferredBodyFontSize) ?? (float)1.0;
			return size * Math.Min(originalScale, FeatureConfiguration.Font.MaximumTextScaleFactor);
```